hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -567,13 +579,14 @@ int MDMParser::waitFinalResp(_CALLBACKPTR cb /* = NULL*/,
                         HAL_NET_notify_dhcp(false);
                     }
                 } else {
-                    // +CREG|CGREG: <n>,<stat>[,<lac>,<ci>[,AcT[,<rac>]]] // reply to AT+CREG|AT+CGREG
-                    // +CREG|CGREG: <stat>[,<lac>,<ci>[,AcT[,<rac>]]]     // URC
-                    b = (int)0xFFFF; c = (int)0xFFFFFFFF; d = -1;
-                    r = sscanf(cmd, "%31s %*d,%d,\"%x\",\"%x\",%d",s,&a,&b,&c,&d);
+                    // +CREG|CGREG: <n>,<stat>[,<lac>,<ci>[,AcT[,<rac>]]] // reply to AT+CREG|AT+CGREG (2,4,5,6 results)
+                    // +CREG|CGREG: <stat>[,<lac>,<ci>[,AcT[,<rac>]]]     // URC (1,3,4,5 results)
+                    b = (int)0xFFFF; c = (int)0xFFFFFFFF; d = -1; mode = 2;
```

Comment:
A comment why specifically we default to 2 would be appreciated. If there is no specific reason, we should probably default to something like `-1`.

Reply:
-1 would be even safer... so I'll go that route.